Fixed indentation including issues with "=" normal mode command.

bak-t opened the pull request:
These changes should fix issue #27. I replaced _smartindent_ with fine tuned _cindent_.

bak-t committed:
Fixed indentation including issues with "=" normal mode command.

heaths reviewed (commented):
Thank you for the PR! I'm doing some manual testing, and found automatic one-space indentation in the following scenario:

<img width="70" alt="image" src="https://github.com/PProvost/vim-ps1/assets/1532486/ab6aad5e-bf65-4a7d-88d9-ea9520b4be04">

Without your changes, everything stays on the previous line's column. Reading about these `cindent`-related options - which I admit in 30 some years I haven't really played with much - it seems may be related to the `cinkeys` options.

Thoughts? I tried to think and test a bunch of indent-related tasks and, for the most part, this does seem to work better, like auto-indenting parameters in a `param()` block.

bak-t committed:
Additional indentation improvements:

 - Improved indentation of comments.
 - Simplified option for unclosed parentheses.

bak-t commented:
I was able to slightly improve indentation of comments with `>` normal mode command (for comment started from 1st column only 1 space is used instead of `shiftwidth`).
But the bug you've found in insert mode still persists. I'll try to experiment with `cinkeys`.

heaths commented:
Thanks. I look forward to it. I think this is a big step in the right direction, but worry if I found an arguably minor regression, what other possibly larger regressions have I missed? Hopefully whatever could resolve this issue would mitigate at least some risk of regression elsewhere.

bak-t commented:
It seems that `cinkeys` doesn't help with the issue found or at least I'm unable to find proper values for it.
Also I found that with current fix `=` normal mode command changes indentation of comments back & forth at root scope at each run. For example:
```powershell
# comment
# another comment
  # mistakenly indented comment
```
after `gg=G` looks like:
```powershell
  # comment
  # another comment
# mistakenly indented comment
```
after `.`:
```powershell
# comment
# another comment
  # mistakenly indented comment
```
and so on.

I'll experiment with `indentexpr`. Looks like it's possible to combine it with C-indent by using `cindent` function. So, solution may be perform handcrafted indentation for edge cases & call `cindent` function otherwise. This also could make possible to skip indentation in here-strings.
